lib/_emerge/Scheduler.py

```diff
@@ -1792,6 +1797,30 @@ def _is_work_scheduled(self):
     def _running_job_count(self):
         return self._jobs
 
+    def _can_add_job(self):
+        if not super()._can_add_job():
+            return False
+
+        if self._jobs_tmpdir_space_threshold is not None and hasattr(os, "statvfs"):
+            tmpdir = first_existing(
+                os.path.join(self.settings["PORTAGE_TMPDIR"], "portage")
+            )
+            try:
+                vfs_stat = os.statvfs(tmpdir)
+            except OSError as e:
+                writemsg_level(
+                    f"!!! statvfs('{tmpdir}'): {e}\n",
+                    noiselevel=-1,
+                    level=logging.ERROR,
+                )
+            else:
+                if (
+                    (vfs_stat.f_blocks - vfs_stat.f_bavail) / vfs_stat.f_blocks
+                ) > self._jobs_tmpdir_space_threshold:
+                    return False
```

Comment:
> Though, I personally would probably have implemented absolute threshold parameters for free disk space and inodes first, and not a relative one.

Since ratios have no units, it was simpler to implement this threshold as a ratio. I was thinking we could possibly re-use the same option for absolute thresholds if we add support for parsing units.

Reply:
We could make it possible to distinguish between used space or remaining space, using a plus or minus operator for example.